include/monkey/mk_core/mk_list.h

```diff
@@ -82,6 +82,29 @@ static inline void mk_list_add_after(struct mk_list *_new,
     next->prev = _new;
 }
 
+static inline void mk_list_add_before(struct mk_list *_new,
+                                      struct mk_list *next,
+                                      struct mk_list *head)
+{
+    struct mk_list *prev;
+
+    if (_new == NULL || next == NULL || head == NULL) {
+        return;
+    }
+
+    if ((head->prev == head && head->next == head) /*empty*/||
```

Comment:
Could you use `mk_list_is_empty` here?

Reply:
Thank you for reviewing. I updated the PR.